bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BValueArray.java

```diff
@@ -0,0 +1,573 @@
+            if (size != -1) {
+                this.size = maxArraySize = size;
+            }
+            byteValues = (byte[]) newArrayInstance(Byte.TYPE);
+            super.arrayType = new BArrayType(BTypes.typeByte, size);
+            this.elementType = BTypes.typeByte;
+        } else if (type.getTag() == TypeTags.FLOAT_TAG) {
+            if (size != -1) {
+                this.size = maxArraySize = size;
+            }
+            floatValues = (double[]) newArrayInstance(Double.TYPE);
+            super.arrayType = new BArrayType(BTypes.typeFloat, size);
+            this.elementType = BTypes.typeFloat;
+        } else if (type.getTag() == TypeTags.STRING_TAG) {
+            if (size != -1) {
+                this.size = maxArraySize = size;
+            }
+            stringValues = (String[]) newArrayInstance(String.class);
+            Arrays.fill(stringValues, BLangConstants.STRING_EMPTY_VALUE);
+            super.arrayType = new BArrayType(BTypes.typeString, size);
+            this.elementType = BTypes.typeString;
+        }
+    }
+
+    // -----------------------  get methods ----------------------------------------------------
+
+    public BRefType<?> getRefValue(long index) {
+        rangeCheckForGet(index, size);
+        return refValues[(int) index];
+    }
+
+    public long getInt(long index) {
+        rangeCheckForGet(index, size);
+        return intValues[(int) index];
+    }
+
+    public int getBoolean(long index) {
+        rangeCheckForGet(index, size);
+        return booleanValues[(int) index];
+    }
+
+    public byte getByte(long index) {
+        rangeCheckForGet(index, size);
+        return byteValues[(int) index];
+    }
+
+    public double getFloat(long index) {
+        rangeCheckForGet(index, size);
+        return floatValues[(int) index];
+    }
+
+    public String getString(long index) {
+        rangeCheckForGet(index, size);
+        return stringValues[(int) index];
+    }
+
+    // ----------------------------  add methods --------------------------------------------------
+
+    public void add(long index, BRefType<?> value) {
+        synchronized (this) {
+            if (freezeStatus.getState() != CPU.FreezeStatus.State.UNFROZEN) {
+                handleInvalidUpdate(freezeStatus.getState());
+            }
+        }
+
+        prepareForAdd(index, refValues.length);
+        refValues[(int) index] = value;
+    }
+
+    public void add(long index, long value) {
+        synchronized (this) {
+            if (freezeStatus.getState() != CPU.FreezeStatus.State.UNFROZEN) {
+                handleInvalidUpdate(freezeStatus.getState());
+            }
+        }
+
+        prepareForAdd(index, intValues.length);
+        intValues[(int) index] = value;
+    }
+
+    public void add(long index, int value) {
+        if (elementType.getTag() == TypeTags.INT_TAG) {
+            add(index, (long) value);
+            return;
+        }
+
+        synchronized (this) {
```

Comment:
Seems line # 263-269 is common for all add methods. Shall we move that to a util method?

Reply:
Done.